### PR TITLE
Add in REST findModelArtifact modelVersionID query param

### LINKED ISSUE #80: Query for a single `ModelArtifact` needs a `modelVersionId` parameter
**Is your feature request related to a problem? Please describe.**
Follow-up to https://github.com/opendatahub-io/model-registry/issues/78

**Describe the solution you'd like**
Acceptance criteria:

For REST operation `findModelArtifact` (_Get a ModelArtifact that matches search parameters_),
instead of currently just getting as input:

- `name`
- `externalId`

make it accept for "search parameter" as input:

- `name`
- `modelVersionId`
- (`externald`)

In order for findModelArtifact to locate the exact Artifact beloning to a specific ModelVersion.

Search by more relaxed parameter and criteria (such as, any Artifact named "README.md" in whichever ModelVersion/RegisteredModel) can be done in future iterations for those operations expecting to return a list of Artifacts instead.

**Describe alternatives you've considered**
n/a

**Additional context**
See similarly was done by analogy in #76

### PR DESCRIPTION
**Pull request by tarilabs**
Resolves #80 

## Description
Address acceptance criteria in #80 

## How Has This Been Tested?
Tested with Swagger editor at: https://editor.swagger.io

You can preview with: https://editor.swagger.io/?url=https://raw.githubusercontent.com/opendatahub-io/model-registry/1c8ba43ebb47690a258cce29efd22cf5ea22fa90/api/openapi/model-registry.yaml#operations-ModelRegistryService-findModelArtifact

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by lampajr — DISMISSED** [review]

File: api/openapi/model-registry.yaml [hunk 1]
```diff
@@ -1688,6 +1690,16 @@ components:
         type: string
       in: query
       required: false
+    modelVersionID:
```
Comment (dhirajsb) [inline api/openapi/model-registry.yaml]: As mentioned in #79 we should instead create either `parentContextId` or `parentResourceId` parameter for all parent context based search endpoints.

**Review by dhirajsb — CHANGES_REQUESTED** [review]

File: api/openapi/model-registry.yaml (hunk 1 above)
Comment (tarilabs) [reply api/openapi/model-registry.yaml]: As I understood from this required-change, I folded the 2 parameters into a single parentResourceID parameter for both, with 2f9b54a
👉 https://github.com/opendatahub-io/model-registry/pull/81#event-10753203936

**Review by dhirajsb — APPROVED** [review]
/lgtm

**Review by lampajr — APPROVED** [review]
lgtm

**Comment by tarilabs** [comment]
two approvals, no veto, proceeding to merge

**Pull request merged**